peru/resources/plugins/bzr/bzr_plugin.py

```diff
@@ -0,0 +1,85 @@
+#! /usr/bin/env python3
+
+import os
+import os.path
+import shutil
+import subprocess
+
+
+def bzr(path, *args, capture_output=False):
+    cwd = os.getcwd()
+    os.chdir(path)
+
+    try:
+        command = ['bzr']
+        command.extend(args)
+        stdout = subprocess.PIPE if capture_output else None
+        process = subprocess.Popen(command, stdin=subprocess.DEVNULL,
+                                   stdout=stdout, universal_newlines=True)
+        output, _ = process.communicate()
+        if process.returncode != 0:
+            raise RuntimeError(
+                'Command exited with error code {0}:\n$ {1}\n{2}'.format(
+                    process.returncode,
+                    ' '.join(command),
+                    output))
+        return output
+    finally:
+        os.chdir(cwd)
+
+
+def clone_if_needed(bzr_path, url):
+    if os.path.exists(os.path.join(bzr_path, '.bzr')):
+        return False
+    try:
+        bzr(bzr_path, 'branch', '--use-existing-dir', url, '.',
+            capture_output=True)
+        return True
+    except:
+        shutil.rmtree(bzr_path)
```

Comment:
We used to do this for git and hg, but we *just* got rid of it in 142181d5bc4a452877aa96139436ff12d995ee19. Git and hg will both delete the clone directory if the clone fails, so `rmtree` was running into errors and just confusing the stacktrace. Do you know if bzr has the same behavior?